### ISSUE TITLE
Why I can't download from HTTPS sites?

### ISSUE THREAD

**Issue by embede12** [ziahamza/webui-aria2#537]
Hi, do I need something extra to have possibility for downloading stuff from HTTPS url's? HTTP works fine.

**Comment by no1xsyzy** [comment]
For lack of information, I have some fuzzy guesses.

1. Check the computer aria2 runs on. It should has all ca-certificates installed;
2. Some censorship stopped you from downloading from particular HTTPS sites;
3. That site just refused your request to download;
4. You typed wrong;
5. It is not actually possible to download without proper header, but your ISP cached the request and response wrongly.

**Comment by embede12** [comment]
Hi! Thanks for your reply, I'm very glad.

> For lack of information, I have some fuzzy guesses.
Sorry for this, but I was tired after hours spent on searching solves for my issue.
> 
>     1. Check the computer aria2 runs on. It should has all ca-certificates installed;
> 
I don't know how I can do that. One thing what I only see when I'm doing research is installing SSL CA for Apache. I think I don't need Apache to have possibility for download files.
But I think, here is the problem. I found somewhere a option, what I can write after URL, I mean `--check-certificate=false`. When I type this, download starting. When I'm trying download without this, I get this error:

`aria2c -x 5 "https://xt7.pl/pobieraj/13da0d611da32a05225431b516f5e919/Sex.Education.2020.PL.S02E08.720p.NF.WEB-DL.X264-J.mkv"01/21 07:55:49 [NOTICE] Downloading 1 item(s)01/21 07:55:49 [NOTICE] CUID#7 - Redirecting to https://s4.rapiduservers.net/download/3fa39ef15df5fbfe1a6fc77437ef815bd5aa67a9/Sex.Education.2020.PL.S02E08.720p.NF.WEB-DL.X264-J.mkv01/21 07:55:50 [ERROR] CUID#7 - Download aborted. URI=https://xt7.pl/pobieraj/13da0d611da32a05225431b516f5e919/Sex.Education.2020.PL.S02E08.720p.NF.WEB-DL.X264-J.mkvException: [AbstractCommand.cc:351] errorCode=1 URI=https://s4.rapiduservers.net/download/3fa39ef15df5fbfe1a6fc77437ef815bd5aa67a9/Sex.Education.2020.PL.S02E08.720p.NF.WEB-DL.X264-J.mkv-> [SocketCore.cc:1015] errorCode=1 SSL/TLS handshake failure:  `not signed by known authorities or invalid' `issuer is not known'`


>     2. Some censorship stopped you from downloading from particular HTTPS sites;
> 
I think no.
>     3. That site just refused your request to download;
> 
Nope or yes?
>     4. You typed wrong;
> 
Nope
>     5. It is not actually possible to download without proper header, but your ISP cached the request and response wrongly.
Nope

**Comment by no1xsyzy** [comment]
Yes it is 1. `ca-certifactes` problem. Not about Apache.
TL;DR: You need find a package named like `ca-certifactes` for your distribution and install it. Maybe it is accidentally not referenced correctly, it can be wired by hand with option `--ca-certificate=<FILE>`.
Full explaination: All certificates are signed by another. A signs B means A trusts B, so if you trust A, then you probably would also trust B, and so on. This is called *certificate chains*. So who do you trust first? Root CAs. Without Root CAs, you don't have the basis of trust. On windows, it is included out of box. On Linux, it is a difficult question, every distribution does differently. Maybe out of box, maybe you need to install it with hand.

**Comment by embede12** [comment]
I have a little update, some url's are working but it depends from what site I'm trying to download.
For example:
This url doesn't work: `https://xt7.pl/pobieraj/13da0d611da32a05225431b516f5e919/Sex.Education.2020.PL.S02E08.720p.NF.WEB-DL.X264-J.mkv`
This work: `https://xt7.pl/pobieraj/323e46cfc5fcb661f4981ec26c143fce/Harry.Potter.and.the.Deathly.Hallows.Part.2.2011.PLDUB.BDRip.x264-MAXiM.mkv`
This doesn't work: `https://linkerbot.pl/download/GkxBI900SJ/1The.Witcher.S01E02.2019.PL.DUAL.480p.WEB-DL.x264-MAXiM.mkv`
This work: `https://xt7.pl/pobieraj/1e5231712870e0139918fda5a2f49f78/PlugInGuru.Airwave.V3.and.V%2526hellip%253Brar`
This work: `https://s25.filebit.pl/plik-afed8e1d0ac2dd208ac1b904eea14286/RD/54399311/58199/the.path.s02e04.pl.720p.amzn.webrip.x264.ac3-kit.mkv`
This doesn't work: `https://s6.rapiduservers.net/download/5e130c74571a0f6f0a773d568236a8bcd89948d7/Ready.or.Not.2019.PL.720p.BluRay.x264.AC3-KiT.mkv`

Like you can see, some xt7.pl are working and some not. I didn't found any working url from rapiduservers or from linkerbot. Only some xt7 and all filebit's url's are working well.

I installed ca-certificates but nothing happen.

> In case it is just accidentally not referenced correctly, it can be wired by hand with option --ca-certificate=<FILE>.

I found some certificates in /etc/ssl/certs/ and this is a list of them. 

>  02265526.0
 03179a64.0
 062cdee6.0
 064e0aa9.0
 06dc52d5.0
 080911ac.0
 09789157.0
 0b1b94ef.0
 0bf05006.0
 0c31d5ce
 0c4c9b6c.0
 0f6fa695.0
 106f3e4d.0
 116bf586.0
 128805a3.0
 157753a5.0
 1636090b.0
 18856ac4.0
 1d3472b9.0
 1e08bfd1.0
 1e09d511.0
 244b5494.0
 2ae6433e.0
 2b349938.0
 2c11d503.0
 2c543cd1.0
 2e4eed3c.0
 2e5ac55d.0
 32888f65.0
 349f2832.0
 3513523f.0
 3bde41ac.0
 3e44d2f7.0
 3e45d192.0
 40193066.0
 4042bcee.0
 40547a79.0
 4304c5e5.0
 451b5485.0
 480720ec.0
 48bec511.0
 4a6481c9.0
 4bfab552.0
 4f316efb.0
 5273a94c.0
 5443e9e3.0
 54657681.0
 57bcb2da.0
 5a4d6896.0
 5ad8a5d6.0
 5c44d531.0
 5cd81ad7.0
 5d3033c5.0
 5f15c80c.0
 607986c7.0
 608a55ad.0
 6410666e.0
 653b494a.0
 6b99d060.0
 6d41d539.0
 6fa5da56.0
 706f604c.0
 749e9e03.0
 75d1b2ed.0
 76cb8f92.0
 76faf6c0.0
 7719f463.0
 7992b8bb.0
 7aaf71c0.0
 7d0b38bd.0
 7f3d5d1d.0
 812e17de.0
 8160b96c.0
 87229d21.0
 8867006a.0
 8cb5ee0f.0
 8d86cdd1.0
 9168f543.0
 930ac5d2.0
 93bc0acc.0
 988a38cb.0
 9c2e7d30.0
 9c8dfbd4.0
 9d04f354.0
 9f0f5fd6.0
 a760e1bd.0
 a94d09e5.0
 ACCVRAIZ1.pem
 AC_RAIZ_FNMT-RCM.pem
 Actalis_Authentication_Root_CA.pem
 ad088e1d.0
 AddTrust_External_Root.pem
 aee5f10d.0
 AffirmTrust_Commercial.pem
 AffirmTrust_Networking.pem
 AffirmTrust_Premium_ECC.pem
 AffirmTrust_Premium.pem
 Amazon_Root_CA_1.pem
 Amazon_Root_CA_2.pem
 Amazon_Root_CA_3.pem
 Amazon_Root_CA_4.pem
 Atos_TrustedRoot_2011.pem
 Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.pem
 b0e59380.0
 b1159c4c.0
 b1b8a7f3.0
 b204d74a.0
 b66938e9.0
 b727005e.0
 b7a5b843.0
 ba89ed3b.0
 Baltimore_CyberTrust_Root.pem
 Buypass_Class_2_Root_CA.pem
 Buypass_Class_3_Root_CA.pem
 c01cdfa2.0
 c089bbbd.0
 c0fb5232
 c0fb5232.0
 c0ff1f52.0
 c28a8a30.0
 c47d9980.0
 ca6e4ad9.0
 ca-certificates.crt
 CA_Disig_Root_R2.pem
 cbf06781.0
 cc450945.0
 cd58d51e.0
 cd8c0d63.0
 ce5e74ef.0
 Certigna.pem
 Certinomis_-_Root_CA.pem
 Certplus_Class_2_Primary_CA.pem
 Certplus_Root_CA_G1.pem
 Certplus_Root_CA_G2.pem
 certSIGN_ROOT_CA.pem
 Certum_Trusted_Network_CA_2.pem
 Certum_Trusted_Network_CA.pem
 CFCA_EV_ROOT.pem
 Chambers_of_Commerce_Root_-_2008.pem
 Comodo_AAA_Services_root.pem
 COMODO_Certification_Authority.pem
 COMODO_ECC_Certification_Authority.pem
 COMODO_RSA_Certification_Authority.pem
 Cybertrust_Global_Root.pem
 d4dae3dd.0
 d6325660.0
 d7e8dc79.0
 d853d49e.0
 dd8e9d41.0
 de6d66f3.0
 def36a68.0
 Deutsche_Telekom_Root_CA_2.pem
 DigiCert_Assured_ID_Root_CA.pem
 DigiCert_Assured_ID_Root_G2.pem
 DigiCert_Assured_ID_Root_G3.pem
 DigiCert_Global_Root_CA.pem
 DigiCert_Global_Root_G2.pem
 DigiCert_Global_Root_G3.pem
 DigiCert_High_Assurance_EV_Root_CA.pem
 DigiCert_Trusted_Root_G4.pem
 DST_Root_CA_X3.pem
 D-TRUST_Root_Class_3_CA_2_2009.pem
 D-TRUST_Root_Class_3_CA_2_EV_2009.pem
 e113c810.0
 e18bfb83.0
 e2799e36.0
 e36a6752.0
 e73d606e.0
 e8de2f56.0
 EC-ACC.pem
 ee64a828.0
 EE_Certification_Centre_Root_CA.pem
 eed8c118.0
 ef954a4e.0
 Entrust.net_Premium_2048_Secure_Server_CA.pem
 Entrust_Root_Certification_Authority_-_EC1.pem
 Entrust_Root_Certification_Authority_-_G2.pem
 Entrust_Root_Certification_Authority.pem
 ePKI_Root_Certification_Authority.pem
 E-Tugra_Certification_Authority.pem
 f060240e.0
 f081611a.0
 f0c70a8d.0
 f30dd6ad.0
 f3377b1b.0
 f387163d.0
 f39fc864.0
 fc5a8f99.0
 fe8a2cd8.0
 ff34af3f.0
 GDCA_TrustAUTH_R5_ROOT.pem
 GeoTrust_Global_CA.pem
 GeoTrust_Primary_Certification_Authority_-_G2.pem
 GeoTrust_Primary_Certification_Authority_-_G3.pem
 GeoTrust_Primary_Certification_Authority.pem
 GeoTrust_Universal_CA_2.pem
 GeoTrust_Universal_CA.pem
 Global_Chambersign_Root_-_2008.pem
 GlobalSign_ECC_Root_CA_-_R4.pem
 GlobalSign_ECC_Root_CA_-_R5.pem
 GlobalSign_Root_CA.pem
 GlobalSign_Root_CA_-_R2.pem
 GlobalSign_Root_CA_-_R3.pem
 Go_Daddy_Class_2_CA.pem
 Go_Daddy_Root_Certificate_Authority_-_G2.pem
 Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.pem
 Hellenic_Academic_and_Research_Institutions_RootCA_2011.pem
 Hellenic_Academic_and_Research_Institutions_RootCA_2015.pem
 Hongkong_Post_Root_CA_1.pem
 IdenTrust_Commercial_Root_CA_1.pem
 IdenTrust_Public_Sector_Root_CA_1.pem
 ISRG_Root_X1.pem
 Izenpe.com.pem
 java
 LuxTrust_Global_Root_2.pem
 Microsec_e-Szigno_Root_CA_2009.pem
'NetLock_Arany_=Class_Gold=_Főtanúsítvány.pem'
 Network_Solutions_Certificate_Authority.pem
 OISTE_WISeKey_Global_Root_GA_CA.pem
 OISTE_WISeKey_Global_Root_GB_CA.pem
 OpenTrust_Root_CA_G1.pem
 OpenTrust_Root_CA_G2.pem
 OpenTrust_Root_CA_G3.pem
 QuoVadis_Root_CA_1_G3.pem
 QuoVadis_Root_CA_2_G3.pem
 QuoVadis_Root_CA_2.pem
 QuoVadis_Root_CA_3_G3.pem
 QuoVadis_Root_CA_3.pem
 QuoVadis_Root_CA.pem
 Secure_Global_CA.pem
 SecureSign_RootCA11.pem
 SecureTrust_CA.pem
 Security_Communication_RootCA2.pem
 Security_Communication_Root_CA.pem
 Sonera_Class_2_Root_CA.pem
 ssl-cert-snakeoil.pem
 SSL.com_EV_Root_Certification_Authority_ECC.pem
 SSL.com_EV_Root_Certification_Authority_RSA_R2.pem
 SSL.com_Root_Certification_Authority_ECC.pem
 SSL.com_Root_Certification_Authority_RSA.pem
 Staat_der_Nederlanden_EV_Root_CA.pem
 Staat_der_Nederlanden_Root_CA_-_G2.pem
 Staat_der_Nederlanden_Root_CA_-_G3.pem
 Starfield_Class_2_CA.pem
 Starfield_Root_Certificate_Authority_-_G2.pem
 Starfield_Services_Root_Certificate_Authority_-_G2.pem
 SwissSign_Gold_CA_-_G2.pem
 SwissSign_Silver_CA_-_G2.pem
 SZAFIR_ROOT_CA2.pem
 Taiwan_GRCA.pem
 TeliaSonera_Root_CA_v1.pem
 thawte_Primary_Root_CA_-_G2.pem
 thawte_Primary_Root_CA_-_G3.pem
 thawte_Primary_Root_CA.pem
 TrustCor_ECA-1.pem
 TrustCor_RootCert_CA-1.pem
 TrustCor_RootCert_CA-2.pem
 Trustis_FPS_Root_CA.pem
 T-TeleSec_GlobalRoot_Class_2.pem
 T-TeleSec_GlobalRoot_Class_3.pem
 TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.pem
 TÜRKTRUST_Elektronik_Sertifika_Hizmet_Sağlayıcısı_H5.pem
 TWCA_Global_Root_CA.pem
 TWCA_Root_Certification_Authority.pem
 USERTrust_ECC_Certification_Authority.pem
 USERTrust_RSA_Certification_Authority.pem
 Verisign_Class_3_Public_Primary_Certification_Authority_-_G3.pem
 VeriSign_Class_3_Public_Primary_Certification_Authority_-_G4.pem
 VeriSign_Class_3_Public_Primary_Certification_Authority_-_G5.pem
 VeriSign_Universal_Root_Certification_Authority.pem
 Visa_eCommerce_Root.pem
 XRamp_Global_CA_Root.pem

**Comment by no1xsyzy** [comment]
Sorry I should admit I might be wrong. If it had been the root certificate problem, no https would have worked. However, `not signed by known authorities` should be certificate problem.
Maybe it is about a wrong configuration of `xt7.pl`. The maintainer of that site might be using some bad configuration. The worse situation is that there is some script kiddie MITM-ing you but they f*cked up and only jammed HTTPS without intervening it.
Anyway, to debug this, it requires more skills, which I am not able to teach, even if I am able to debug by myself. To be short, `curl -v <url> --output <localpath>` will print a lot. When it failed, you can see what happens.
What's more, it must be something like `--ca-certificate=something.pem`. [Reference](https://aria2.github.io/manual/en/html/aria2c.html#cmdoption-ca-certificate).

**Comment by embede12** [comment]
> Anyway, to debug this, it requires more skills, which I am not able to teach, even if I am able to debug by myself.

Ok, you mean, we can't do anything more to fix that?
It's dangerous to still using option `--check-certificate=false`?

**Comment by no1xsyzy** [comment]
Nothing to do through this time-decoupled discussion. An instant communication method is required.
However, we are at different timezone, so it is hard to find a time to do.
If one more suggestion, I think asking tech support from `xt7.pl` will be better.
